packit_service/worker/jobs.py

```diff
@@ -232,26 +231,28 @@ def process_jobs(self, event: Event) -> Dict[str, HandlerResults]:
                 event, event.project, config=self.config
             ):
                 for job_config in job_configs:
-                    handlers_results[job_config.type.value] = HandlerResults(
+                    processing_results[job_config.type.value] = TaskResults(
                         success=False, details={"msg": "Account is not whitelisted!"}
                     )
-                return handlers_results
+                return processing_results
 
             # we want to run handlers for all possible jobs, not just the first one
             for job_config in job_configs:
-                logger.debug(f"Running handler: {str(handler_kls)} for {job_config}")
-                event_dict = event.get_dict()
-                handler = handler_kls(
-                    package_config=event.package_config,
-                    job_config=job_config,
-                    data=EventData.from_event_dict(event_dict),
+                send_handler_task(
+                    task_name=handler_kls.task_name, event=event, job=job_config
```

Comment:
Not questioning your design, just wondering:
- have you seen [group primitive of Celery](https://docs.celeryproject.org/en/stable/userguide/canvas.html#groups)?
- would using it give us any benefit?
- could it perhaps help us to get rid of our `TaskResults`, by using their [Group Results](https://docs.celeryproject.org/en/stable/userguide/canvas.html#group-results)?

Reply:
I have already seen it, but I will recheck it, if it can give us some benefit

Reply:
So, I do not consider Group Results really beneficial, we can get the success/failure (in Celery context) of the tasks by waiting for finishing the tasks and waiting for tasks within a task is not recommended in docs. In our `TaskResults` we can add additional info which describes the task. The difference in using group and the current implementation would be that group (tasks executing matching jobs) "Creates a group of tasks to be executed in parallel."  and in the current implementation for all matching jobs is `send_task` executed.

Reply:
> we can get the success/failure (in Celery context) of the tasks by waiting for finishing the tasks and waiting for tasks within a task is not recommended in docs

Are you talking about the current state or about the possible 'group workflow'?

> In our `TaskResults` we can add additional info which describes the task.

I'm sorry, I confused `TaskResults`with `task_results`/`jobs_results`. I think you can ignore me then.

> group (tasks executing matching jobs) "Creates a group of tasks to be executed in parallel."

and I was thinking that that might be what we actually want, i.e. let Celery run them in parallel and collect results once all are done. But maybe not :-)

> in the current implementation for all matching jobs is `send_task` executed

yes, initially we were using `send_task`, because we were sending the task (there was only one) from service to workers and because the task was not registered in the service we were calling it by name (see Tip [here](https://docs.celeryproject.org/en/stable/userguide/calling.html#example)). But now when we send task from worker to worker(s), we can use any other mechanism (like described in [Groups](https://docs.celeryproject.org/en/stable/userguide/canvas.html#groups)), can't we?

Anyway, I don't want to argue here, I'll let this on you since you're now the Celery guru and my map of the related code is heavily outdated. I just want to suggest that when we are doing such a big change we can investigate whether there already isn't a better mechanism in Celery.

Reply:
> Are you talking about the current state or about the possible 'group workflow'?

About 'group workflow', in the current after sending the handlers tasks we end the processing task.

> and I was thinking that that might be what we actually want, i.e. let Celery run them in parallel and collect results once all are done. But maybe not :-)

Yes, it would be the best if we could collect the results once all are done, but if I understand the docs correctly, we would need to wait in the processing task for finishing all of the tasks.

> yes, initially we were using send_task, because we were sending the task (there was only one) from service to workers and because the task was not registered in the service we were calling it by name (see Tip here). But now when we send tasks from worker to worker(s), we can use any other mechanism (like described in Groups), can't we?

thanks for the explanation, now I understand it...so yes, then we can use other method

> Anyway, I don't want to argue here, I'll let this on you since you're now the Celery guru and my map of the related code is heavily outdated. I just wanted to suggest that when we are doing such a big change we can investigate whether there already isn't a better mechanism in Celery.

No arguing, I am glad for your feedback